### PR TITLE
Tests for ASes skipping rate limits

### PR DESCRIPTION
**Pull request by illicitonion**

### CODE REVIEW

File: tests/60app-services/02ghost.pl [hunk 1]
```diff
@@ -87,6 +87,72 @@
...
+   requires => [ as_ghost_fixture(), $main::AS_USER,
+                     room_fixture( requires_users => [ $user_fixture ] ),
+                qw( can_receive_room_message_locally )],
+
+   do => sub {
+      my ( $ghost, $as_user, $room_id ) = @_;
+
+      Future->needs_all(
+         ( map {
+            await_as_event( "m.room.member" )->then( sub { Future->done( 1 ); } ),
```
Comment (leonerd) [inline tests/60app-services/02ghost.pl]: There's a neater form here

    $f->then_done(1);

is a shortcut for

    $f->then( sub { Future->done(1) } );

File: tests/60app-services/02ghost.pl [hunk 2]
```diff
@@ -87,6 +87,72 @@
...
+            params => {
+               user_id => $as_user->user_id,
+            },
+
+            content => {},
+         ),
+      )->then( sub {
+         Future->needs_all(
+            ( map {
+               await_as_event( "m.room.message" )->then( sub { Future->done( 1 ); } )
```
Comment (leonerd) [inline tests/60app-services/02ghost.pl]: Likewise `then_done` here

File: tests/60app-services/02ghost.pl [hunk 3]
```diff
@@ -87,6 +87,72 @@
...
+            } 0 .. 300 ),
```
Comment (leonerd) [inline tests/60app-services/02ghost.pl]: There's an off-by-one in this code. The range `0 .. 300` is inclusive at both ends, so has 301 elements. Similarly in the message sending you send 151 messages from each user; a total of 302.

I'd suggest using the ranges `1 .. 300` and `1 .. 150` here

Comment (illicitonion) [inline tests/60app-services/02ghost.pl]: Done

File: tests/60app-services/02ghost.pl (hunk 2 above)
Comment (illicitonion) [inline tests/60app-services/02ghost.pl]: Done

File: tests/60app-services/02ghost.pl (hunk 1 above)
Comment (illicitonion) [inline tests/60app-services/02ghost.pl]: Done

**Comment by NegativeMjark** [comment]
Other than not passing LGTM

**Pull request closed without merging**